### PR TITLE
Several improvements on Liqoctl

### PR DESCRIPTION
**Pull request by palexster**
# Description

This commit introduces: 
- [x] a provider to add the [kind](https://kind.sigs.k8s.io/) provider 
- [x] the possibility to dump the values.yaml files instead of trigger the installation.
- [x] add the possibility to set-up the cluster labels from liqoctl

# How Has This Been Tested?

- [x] Manual
- [x] Unit

### CODE REVIEW

File: pkg/liqoctl/install/utils/map.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	for k := range baseMap {
+		keys[i] = k
+		i++
+	}
+
+	for k := range patchMap {
+		keys[i] = k
+		i++
+	}
+	return keys
```
Comment (aleoli) [inline pkg/liqoctl/install/utils/map.go:52]: here there can be a lot of repeated values, I can suggest something like that:

```suggestion
	keys := make(map[string]struct{})
	for k := range baseMap {
		keys[k] = struct{}
	}

	for k := range patchMap {
		keys[k] = struct{}
	}
	
	keysArr := make([]string, len(keys))
	i := 0
	for k := range keys {
		keys[i] = k
		i++
	}
	return keysArr
```

**Review by aleoli — APPROVED** [review]

File: pkg/liqoctl/install/kubeadm/types.go [hunk 2]
```diff
@@ -17,9 +17,9 @@ var kubeControllerManagerLabels = map[string]string{"component": "kube-controlle
...
-	apiServer   string
-	config      *rest.Config
-	podCIDR     string
-	serviceCIDR string
-	k8sClient   kubernetes.Interface
+	APIServer   string
+	Config      *rest.Config
+	PodCIDR     string
+	ServiceCIDR string
+	K8sClient   kubernetes.Interface
```
[the remark this answers is not in the document]
Comment (palexster) [reply pkg/liqoctl/install/kubeadm/types.go:24]: They are used in the Kind provider struct.

File: cmd/liqoctl/cmd/install.go [hunk 3]
```diff
@@ -23,6 +23,14 @@ func newInstallCommand(ctx context.Context) *cobra.Command {
 	installCmd.Flags().StringP("provider", "p", "kubeadm", "Select the cluster provider type")
 	installCmd.Flags().IntP("timeout", "t", 600, "Configure the timeout for the installation process in seconds (default: 600)")
 	installCmd.Flags().StringP("version", "", "", "Select the Liqo version (default: latest stable release)")
+	installCmd.Flags().BoolP("devel", "", false,
+		"Enable use development versions, too. Equivalent to version '>0.0.0-0'. If --version is set, this is ignored")
+	installCmd.Flags().BoolP("only-output-values", "", false, "Generate a values file for further customization")
```
[the remark this answers is not in the document]
Comment (palexster) [reply cmd/liqoctl/cmd/install.go:28]: I agree that ```only-output-values``` is ugly, but I wanted to stress the fact that the command will just print the output values without performing any installation. What about ```print-output-values```?

[the remark this answers is not in the document]
Comment (aleoli) [reply cmd/liqoctl/cmd/install.go:28]: yes, or `print-values` only

[the remark this answers is not in the document]
Comment (palexster) [reply cmd/liqoctl/cmd/install.go:28]: print-chart-values?

**Review by alacuku — APPROVED** [review]

**Pull request merged**